infrastructure/ansible/roles/traffic_opsdb/defaults/main.yml

```diff
@@ -15,9 +15,9 @@
 install_traffic_opsdb: false
 initialize_traffic_opsdb: false
 
-postgresql_server_pkg_name: "{% if postgresql_version is defined and postgresql_version != omit %}{{ 'postgresql96-server-'+postgresql_version }}{% else %}postgresql96-server{% endif %}"
-postgresql_client_pkg_name: "{% if postgresql_version is defined and postgresql_version != omit %}{{ 'postgresql96-'+postgresql_version }}{% else %}postgresql96{% endif %}"
-postgresql_devel_pkg_name: "{% if postgresql_version is defined and postgresql_version != omit %}{{ 'postgresql96-devel-'+postgresql_version }}{% else %}postgresql96-devel{% endif %}"
+postgresql_server_pkg_name: "{% if pg_major_version | int < 10 %}postgresql{{pg_major_version}}{{pg_minor_version}}-server-{{postgresql_version}}{% else %}postgresql{{pg_major_version}}-server-{{postgresql_version}}{% endif %}"
+postgresql_client_pkg_name: "{% if pg_major_version | int < 10 %}postgresql{{pg_major_version}}{{pg_minor_version}}-{{postgresql_version}}{% else %}postgresql{{pg_major_version}}-{{postgresql_version}}{% endif %}"
+postgresql_devel_pkg_name: "{% if pg_major_version | int < 10 %}postgresql{{pg_major_version}}{{pg_minor_version}}-devel-{{postgresql_version}}{% else %}postgresql{{pg_major_version}}-devel-{{postgresql_version}}{% endif %}"
```

Comment:
If postgresql_version is undefined here, it will fail even though the major/minor are set by default. Notice in the original the hyphen and more specific version are missing including the build are missing entirely.

Reply:
updated to address undefined postgresql_version